src/DopplerVelocityLog.cc

```diff
@@ -1731,12 +1731,34 @@ namespace gz
               samplePointInSensorFrame;
 
           // Transform sample point to the environmental data frame
-          const gz::math::Vector3d samplePointInDataFrame =
+          const std::optional<gz::math::CoordinateVector3>
+              samplePointInDataFrameCoordVec =
               this->worldState->origin.PositionTransform(
-                  samplePointInWorldFrame,
+                  gz::math::CoordinateVector3::Metric(samplePointInWorldFrame),
```

Comment:
@iche033 Technically, if somebody used LOCAL for reference, this will probably silently change the behavior. It is possible this would go completely without a build warning if the value is read from SDF. But I haven't worked with DVL, so I don't know if setting LOCAL would make sense...

Reply:
I'm not very familiar with it as well. I looked at the possible frames that gz-sim eventually passes to DVL, and I think they are: `GLOBAL`, `SPHERICAL`, and `ECEF` (from the [EnvironentPreload](https://github.com/gazebosim/gz-sim/blob/d662000273ae88e041ee30ec8d24b5e3beed6db7/src/systems/environment_preload/EnvironmentPreload.cc#L171-L204) system, and [EnvironmentLoader](https://github.com/gazebosim/gz-sim/blob/gz-sim8/src/gui/plugins/environment_loader/EnvironmentLoader.cc#L85-L87) GUI plugin). Maybe @arjo129 knows more about this can chime in?

Reply:
@arjo129 what do you think?